### ISSUE TITLE
Revert 'requireSpacesInsideArrayBrackets' change in 1.10

### ISSUE THREAD

**Issue by necolas** [jscs-dev/node-jscs#956]
Since https://github.com/jscs-dev/node-jscs/issues/923 was closed for comment…

Please could you put out a release that reverts the issue reported in #923 before working on the resolution described in that ticket. A standard `npm install --save-dev jscs` sets the range as `^1.9.0` in a project's `package.json`, so anyone running `npm install` today (or travis ci builds) is installing `1.10.0` and reporting 100s/1000s of errors where there were none before.

**Comment by mikesherov** [comment]
@necolas, thanks for the feedback as always. 

We do mention what our criteria for release is here: https://github.com/jscs-dev/node-jscs/blob/master/OVERVIEW.md#versioning--semver and felt that the changes to `requireSpacesInsideArrayBrackets` was a bug fix, and therefore warranted a minor release.

With that said, yes, we'll work on reverting the behavior first.

**Comment by necolas** [comment]
Thanks :)

**Comment by markelog** [comment]
@necolas 

> reporting 100s/1000s of errors where there were none before.

if you are still going to use caret, this situation will happen again, we're trying really hard to make patch releases without additional error reports specifically for this reason, so if you don't want this kind of mishaps in the future - use `~`.

**Comment by mikesherov** [comment]
@markelog I'm on this right now FYI. Should have PR shortly.

**Comment by necolas** [comment]
> if you are still going to use caret, this situation will happen again

Do you plan on changing rule behaviour for minor releases?

**Comment by markelog** [comment]
> Do you plan on changing rule behaviour for minor releases?

Nope, but we will try our best to fix known bugs and if we fix them, you might get unexpected 1000 errors from nowhere in your CI-run

**Comment by necolas** [comment]
I understand. Happy for that to happen if it catches issues :)

**Comment by mikesherov** [comment]
@necolas just to be clear, we put bug fixes that cause less errors to be detected in patch, bug fixes that more errors to be detected in minor, and rule meaning changes into major.

One man's rule meaning change is another man's bug fix. In this case, we considered it a bug fix when clearly people were using it a different way. Hopefully it won't happen again, but we can't guarantee that. 

TLDR: `~` will always get bug fixes unless they report more errors. `^` will get all bug fixes, even if they break your build.

**Comment by necolas** [comment]
Thanks for the helpful suggestions and clarification.

**Comment by mikesherov** [comment]
@markelog ping for review and advice on how to address in presets.
